### ISSUE TITLE
training mask overwritten when used

### ISSUE THREAD

**Issue by shackett** [napistu/napistu-torch#41]
This seems like an MPS bug

```python
sum_before_indexing = data.train_mask.sum().item()
        
# Always use training edges for message passing
# WORKAROUND: MPS backend bug - boolean indexing corrupts the mask
# Create a detached clone to prevent in-place modification
train_mask_safe = data.train_mask.clone().detach()
supervision_edges = data.edge_index[:, train_mask_safe]

# Check after indexing
sum_after_indexing = data.train_mask.sum().item()
if sum_before_indexing != sum_after_indexing:
    raise RuntimeError(
        f"MASK CORRUPTED BY INDEXING OPERATION!\n"
        f"Before: {sum_before_indexing}\n"
        f"After: {sum_after_indexing}\n"
        f"This line: data.edge_index[:, data.train_mask]"
    )
```

This can throw an error sporadically if just using data.train_mask directly.

This issue seems to be cropping up in some other places resulting in mismatched dimensions.

Additional safeguards/validation is probably needed elsewhere like:

```python
train_mask_safe = self.data.train_mask.clone().detach()
train_edge_indices = torch.where(train_mask_safe)[0]
```
